presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusQueryResponseParse.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import io.prestosql.spi.PrestoException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.prestosql.plugin.prometheus.PrometheusErrorCode.PROMETHEUS_PARSE_ERROR;
+
+public class PrometheusQueryResponseParse
+{
+    private final InputStream response;
+    private boolean status;
+
+    private String error;
+    private String errorType;
+    private List<String> warnings; //TODO not parsing warnings for now
+    private String resultType;
+    private String result;
+    private List<PrometheusMetricResult> results;
+    private PrometheusTimeSeriesValue stringOrScalarResult;
+
+    public PrometheusQueryResponseParse(InputStream response)
+            throws IOException
+    {
+        this.response = response;
+        parse();
+    }
+
+    private boolean parse()
+            throws IOException
+    {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Can we separate parsing from DTO? Can we use regular `jsonCodec` here? Or if not, can we use custom `ObjectMapper` that creates only fields that you care?

Reply:
We are using `ObjectMapper` down below this bit of custom parsing code. The custom parse is handling the Prometheus "header" so to speak and ObjectMapper is handling the objects as desired. I supposed we could make a more complex object to handle the header too, but it seems like it will be more convoluted to handle the error and (TODO) warning in that scenario.